proxy/nginx/lualib/router.lua

```diff
@@ -30,12 +31,21 @@ local key = prefix .. ":routes:" .. host
 
 -- Try to get target for host
 res, err = red:get(key)
+
+-- Exit if route could not be read
+if err then
+    ngx.log(ngx.ERR, "error reading route: " .. err)
+    ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
```

Comment:
If I understood correctly, this is an error while talking to redis. But I agree that exiting here is not a good solution. A better solution would be having a second, long lived cache to fetch routes from when communication to redis fails (the default cache only lasts 5 seconds), but I'm open to suggestions.